pyuvdata/uvdata/mwa_corr_fits.py

```diff
@@ -31,6 +33,232 @@ def input_output_mapping():
+                / sighat_vector_prime(guess[inds])
+            )
+            inds = np.where(np.abs(sighat_vector(guess) - sighat) > 1e-10)[0]
+        sighat_arr[nonzero_inds] = guess
+
+    return sighat_arr
+
+
+def van_vleck_crosses_int(k_arr, sig1_arr, sig2_arr, cheby_approx):
+    """
+    Use Newton's method to solve the inverse of corrcorrect_simps.
+
+    For an explanation of the Van Vleck corrections used and their implementation
+    in this code, see the memos at
+    https://github.com/EoRImaging/Memos/blob/master/PDFs/007_Van_Vleck_A.pdf and
+    https://github.com/EoRImaging/Memos/blob/master/PDFs/008_Van_Vleck_B.pdf
+
+    Parameters
+    ----------
+    k_arr : numpy array
+        Array of quantized kappa to be corrected.
+    sig1_arr : numpy array
+        Array of sigma inputs corresponding to antenna 1.
+    sig2_arr: numpy array
+        Array of sigma inputs corresponding to antenna 2.
+    cheby_approx : bool
+        Flag to warn if chebyshev approximation is being used.
+    """
+    nonzero_inds = np.where((k_arr != 0) & (sig1_arr != 0) & (sig2_arr != 0))[0]
+    if len(nonzero_inds) > 0.0:
+        if cheby_approx:
+            warnings.warn(
+                str(len(nonzero_inds))
+                + " values are being corrected with the van vleck integral"
+            )
+        neg_inds = np.where(k_arr < 0.0)[0]
+        khat = np.abs(k_arr[nonzero_inds])
+        sig1 = sig1_arr[nonzero_inds]
+        sig2 = sig2_arr[nonzero_inds]
+        x0 = khat / (sig1 * sig2)
+        corr = corrcorrect_simps(x0, sig1, sig2) - khat
+        x0 = x0 - (corr / corrcorrect_vect_prime(x0, sig1, sig2))
+        inds = np.where(np.abs(corr) > 1e-8)[0]
+        while len(inds) != 0:
+            corr = corrcorrect_simps(x0[inds], sig1[inds], sig2[inds]) - khat[inds]
+            x0[inds] = x0[inds] - (
+                corr / corrcorrect_vect_prime(x0[inds], sig1[inds], sig2[inds])
+            )
+            inds2 = np.where(np.abs(corr) > 1e-8)[0]
+            inds = inds[inds2]
+        k_arr[nonzero_inds] = x0 * sig1 * sig2
+        k_arr[neg_inds] = np.negative(k_arr[neg_inds])
+
+    return k_arr
+
+
+def van_vleck_crosses_cheby(
+    khat,
+    sig1,
+    sig2,
+    broad_inds,
+    rho_coeff,
+    sv_inds_right1,
+    sv_inds_right2,
+    ds1,
+    ds2,
+    cheby_approx,
+):
+    """
+    Compute a chebyshev approximation of corrcorrect_simps.
+
+    Uses a bilinear interpolation to find chebyshev coefficients. Assumes distance
+    between points of interpolation grid is 0.01. If sig1 or sig2 falls outside
+    the interpolation grid, the corresponding values are corrected using
+    van_vleck_crosses_int.
+
+    For an explanation of the Van Vleck corrections used and their implementation
+    in this code, see the memos at
+    https://github.com/EoRImaging/Memos/blob/master/PDFs/007_Van_Vleck_A.pdf and
+    https://github.com/EoRImaging/Memos/blob/master/PDFs/008_Van_Vleck_B.pdf
```

Comment:
Do you actually have anything about the chebyshev approximation in these memos? If not, we should probably get it into one of them.

Reply:
It isn't in there yet and it should be. I can work on that.

Reply:
I think it is there now.